snowfall/decoding/rescore.py

```diff
@@ -142,31 +144,36 @@ def rescore(lats: k2.Fsa,
     indices2 = torch.argsort(len_per_path, descending=True)
     second_pass_supervision_segments = second_pass_supervision_segments[
         indices2]
-    # note that path_to_seq is not changed!
-    # no need to modify second_pass_out
+    # Note that path_to_seq is not changed!
+    # No need to modify second_pass_out
 
     num_repeats_float = k2.ragged.RaggedFloat(
         num_repeats.shape(),
         num_repeats.values().to(torch.float32))
     path_weight = k2.ragged.normalize_scores(num_repeats_float,
                                              use_log=False).values
-    path_weight = path_weight[indices2]
 
     second_pass_dense_fsa_vec = k2.DenseFsaVec(
         second_pass_out, second_pass_supervision_segments)
 
     second_pass_lattices = k2.intersect_dense_pruned(
-        decoding_graph, second_pass_dense_fsa_vec, 20.0, 7.0, 30, 10000)
+        decoding_graph, second_pass_dense_fsa_vec, 20.0, 7.0, 30, 20000)
```

Comment:
This line **SOMETIMES** fails the following check
https://github.com/k2-fsa/k2/blob/171ddc17509c0de5e9f7dcc4efeed9c712830233/k2/csrc/fsa_utils.cu#L699

```cpp
K2_CHECK_GT(dest_state, state_idx01);
```

I cannot find the reason since it does not always happen. @danpovey  Do you have any suggestions?

Reply:
That is essentially asserting that the FSA is top-sorted and acyclic.  We'd have to consider where the FSA came from... what was the call stack?

Reply:
Will try to get the call stack.